### ISSUE TITLE
OpenGL setting in CMakeList.txt

### ISSUE THREAD

**Issue by scottfurry** [copperspice/copperspice#82]
Building from git source using CMake the following message as appeared:

> CMake Warning (dev) at /usr/share/cmake-3.13/Modules/FindOpenGL.cmake:270 (message):
>  Policy CMP0072 is not set: FindOpenGL prefers GLVND by default when
>  available.  Run "cmake --help-policy CMP0072" for policy details.  Use the
>  cmake_policy command to set the policy and suppress this warning.
>  FindOpenGL found both a legacy GL library:
>    OPENGL_gl_LIBRARY: /usr/lib/libGL.so
>  and GLVND libraries for OpenGL and GLX:
>    OPENGL_opengl_LIBRARY: /usr/lib/libOpenGL.so
>    OPENGL_glx_LIBRARY: /usr/lib/libGLX.so
>
>  OpenGL_GL_PREFERENCE has not been set to "GLVND" or "LEGACY", so for
>  compatibility with CMake 3.10 and below the legacy GL library will be used.
>Call Stack (most recent call first):
>  CMakeLists.txt:298 (find_package)
> This warning is for project developers.  Use -Wno-dev to suppress it.

This can be easily fixed by passing to cmake on the command line: 
`-DOpenGL_GL_PREFERENCE=GLVND`

CMakeLists.txt file probably needs a clearer statement of needed OpenGL level

**Comment by agserm** [comment]
Awesome! Thanks for reporting this. We will take a look at it right away.

**Comment by scottfurry** [comment]
Issue not resolved. Policy may need to be explicitly set.
`
if(POLICY CMP0072)
--
# FindOpenGL prefers GLVND by default when available.
    cmake_policy(SET CMP0072 NEW)
endif()
` 
Reference [here](https://code.i-harness.com/en/docs/cmake~3.12/module/findopengl) gives some details regarding OpenGL policy settings. Choices from link are GVLND or LEGACY.

**Comment by agserm** [comment]
Is it still reporting the exact same warning message as before? I can see from your initial report that you are using CMake 3.13. Opting in to all policies from 3.4.3 to 3.13 seems like it should suppress this warning. I am puzzled as to why this specific policy would need to be set explicitly.

**Comment by scottfurry** [comment]
Same message as before.
The reason for the continued failure could be due to what I posted in last message: change in OpenGL CMake policy. I think it is looking OpenGL would prefer an explicit statement of preference. I think...

The reason for failure after opting into policies could be because opting is for CMake only(? maybe). This is an OpenGL policy setting.

I know people over at wxWidgets went through this recently when someone was building up their CMakeLists.txt file. In the end, they had to put in code similar to snippet above.

Found these links:
https://cmake.org/cmake/help/latest/policy/CMP0072.html
https://cmake.org/cmake/help/latest/module/FindOpenGL.html
